src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate+QuickActions.swift

```diff
@@ -3,13 +3,62 @@
 //
 
 import UIKit
+import AVFoundation
 
 // MARK: - Quick Actions
 
-private enum Shortcut: String {
+enum QuickAction: String {
+	
 	/// General identifier for the 'add diary entry' shortcut action
 	case diaryNewEntry = "de.rki.coronawarnapp.shortcut.diarynewentry"
 	case eventCheckin = "de.rki.coronawarnapp.shortcut.eventcheckin"
+	
+	static var exposureSubmissionFlowTestResult: TestResult?
+	
+	private static var willResignActiveNotification: NSObjectProtocol?
+	
+	static func setup() {
+		guard willResignActiveNotification == nil else {
+			return
+		}
+		willResignActiveNotification = NotificationCenter.default.addObserver(forName: UIApplication.willResignActiveNotification, object: nil, queue: nil) { _ in
+			update()
+		}
+		// setup initial quick actions
+		update()
+	}
+	
+	/// Adds or removes quick actions accoring to the current application state
+	private static func update() {
+		
+		Log.info(#function, log: .ui)
+		
+		let application = UIApplication.shared
+				
+		// No shortcuts if test result positiv
+		if let testResult = exposureSubmissionFlowTestResult, testResult == .positive {
+			application.shortcutItems = nil
+			return
+		}
+
+		// No shortcuts if not onboarded
+		guard let appDelegate = application.delegate as? AppDelegate, appDelegate.store.isOnboarded else {
+			application.shortcutItems = nil
+			return
+		}
+
+		var shortcutItems = [UIApplicationShortcutItem(type: QuickAction.diaryNewEntry.rawValue, localizedTitle: AppStrings.QuickActions.contactDiaryNewEntry, localizedSubtitle: nil, icon: UIApplicationShortcutIcon(templateImageName: "book.closed"))]
+		
+		let status = AVCaptureDevice.authorizationStatus(for: .video)
+		if status == .authorized || status == .notDetermined {
```

Comment:
I like the Idee, but is it synced with POs?

Reply:
I am not sure but it is in sync with the ticket description.